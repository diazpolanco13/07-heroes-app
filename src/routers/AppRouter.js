import React, { useContext } from "react";
import { PrivateRoute } from "./PrivateRoute";
import { BrowserRouter as Router, Switch } from "react-router-dom";
import { LoginScreen } from "../components/login/LoginScreen";
import { DashboardRouter } from "./DashboardRouter";
import { AuthContext } from "../auth/AuthContext";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {
  const { user } = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path="/login"
            component={LoginScreen}
            isAutenticated={user.logged}
          />
          <PrivateRoute
            path="/"
            component={DashboardRouter}
            isAutenticated={user.logged}
          />
        </Switch>
      </div>
    </Router>
  );
};
