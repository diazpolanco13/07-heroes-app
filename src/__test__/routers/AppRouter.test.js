import React from "react";
import "@testing-library/dom";
import { mount } from "enzyme";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from "../../auth/AuthContext";

describe("Prueba del componente <AppRouter />", () => {
  test("Debe mostrar el login sino esta autenticado", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: false,
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    console.log(wrapper.html());
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el componente marvel si esta autenticado", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: "CArlos",
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
