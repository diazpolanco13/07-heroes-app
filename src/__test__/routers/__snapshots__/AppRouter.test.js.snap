// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <AppRouter /> Debe mostrar el login sino esta autenticado 1`] = `
<div>
  <div
    className="container mt-5 w-50 border border-dark rounded p-5 align-middle"
  >
    <form
      action="/examples/actions/confirmation.php"
      method="post"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="inputEmail"
        >
          Email
        </label>
        <input
          className="form-control"
          id="inputEmail"
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="inputPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="inputPassword"
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="checkbox"
      >
        <label>
          <input
            type="checkbox"
          />
           Remember me
        </label>
      </div>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
